server/routes/choropleth.py

```diff
@@ -0,0 +1,215 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines endpoints to support choropleth map."""
+from flask import request
+from flask import jsonify
+from flask import Blueprint
+import numpy as np
+import json
+import services.datacommons as dc
+
+# Defines the map from higher geos to their respective subgeos.
+LEVEL_MAP = {
+    "Country": "AdministrativeArea1",
+    "AdministrativeArea1": "AdministrativeArea2",
+    "AdministrativeArea2": "City"
+}
+
+# All choropleth endpoints are defined with the choropleth/ prefix.
+bp = Blueprint(
+  "choropleth",
+  __name__,
+  url_prefix='/tools/choropleth'
+)
+
+@bp.route('/download')
+def download():
+    """Returns data for geographic subregions for a certain statistical 
+            variable.
+
+    API Params:
+        statVar -> The statistical variable to download, as a string.
+        perCapita -> Whether to return the per-capita value, as a string of a 
+                boolean.
+        geoId -> The currently viewed geography to render, as a string.
+        level -> The subgeographic level to pull and display information for,
+                as a string. Choices: Country, AdministrativeLevel[1/2], City.
+        mdom -> The measurement denominator to use if perCapita is true, as a 
+                string. Defaults to "Count_Person".
+
+
+    API Returns:
+        df -> Returns a dictionary with two entries
+            geoJson -> geoJson format that includes statistical variables info,
+                    geoId, and name for all subregions.
+            _PLOTTING_INFO -> Includes domain range for use in D3 and the
+                    color palette to render with.
+    """
+    # Get required request parameters.
+    requested_stat_var = request.args.get("statVar")
+    if requested_stat_var is None:
+        return jsonify({"error": "Must provide a 'statVar' field!"}, 400)
+    requested_geoId = request.args.get("geoId")
+    if requested_geoId is None:
+        return jsonify({"error": "Must provide a 'geoId' field!"}, 400)
+
+    # If no level is requested then default to one level below current.
+    display_level = request.args.get("level")
+    if display_level is None:
+        requested_geoId_type = dc.get_property_values([requested_geoId],
+                                                "typeOf")[requested_geoId]
+        # TODO(iancostello): Handle a failed function call, e.g., returns None.
+        # TODO(iancostello): Handle the case where display_level is None.
+        for level in requested_geoId_type:
+            if level in LEVEL_MAP:
+                display_level = LEVEL_MAP[level]
+                break
+
+    # Get optional fields.
+    per_capita = request.args.get("perCapita", default=False).lower() in (
```

Comment:
It probably suffices to check that the argument exists. Any value ascribed to it (or nothing) means True.

Reply:
The API may send perCapita=false